x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/register.ts

```diff
@@ -35,14 +35,19 @@ export function registerCspmUsageCollector(
       return true;
     },
     fetch: async (collectorFetchContext: CollectorFetchContext) => {
-      const [
-        indicesStats,
-        accountsStats,
-        resourcesStats,
-        rulesStats,
-        installationStats,
-        alertsStats,
-      ] = await Promise.all([
+      const getPromiseValue = <T>(
```

Comment:
_nit_ you could improve it a bit by not tangling yourself with `results[X]` and working with it by using the following changing `getPromiseValue` to receive a promise and wait for it
So when forming the allSettled request, the naming of the request is right next to it. 
And now `getPromiseValue` name also describes better its logic. 
And maybe `awaitPromiseSafe` describes it even better. It is safe because it won't throw any errors

```ts
      const getPromiseValue = async <T>(
        taskName: CloudSecurityUsageCollectorType,
        promise: Promise<T>
      ) => {
        try {
          const val = await promise;
          return val;
        } catch (error) {
          logger.error(`${taskName} task failed: ${error.message}`);
          logger.error(error.stack);
        }
      };
      const [
        indices,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        accounts_stats,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        resources_stats,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        rules_stats,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        installation_stats,
        // eslint-disable-next-line @typescript-eslint/naming-convention
        alerts_stats,
      ] = await Promise.allSettled([
        getPromiseValue(
          'Indices',
          getIndicesStats(
            collectorFetchContext.esClient,
            collectorFetchContext.soClient,
            coreServices,
            logger
          )
        ),
        getPromiseValue('Accounts', getAccountsStats(collectorFetchContext.esClient, logger)),
        getPromiseValue('Resources', getResourcesStats(collectorFetchContext.esClient, logger)),
        getPromiseValue('Rules', getRulesStats(collectorFetchContext.esClient, logger)),
        getPromiseValue(
          'Installation',
          getInstallationStats(
            collectorFetchContext.esClient,
            collectorFetchContext.soClient,
            coreServices,
            logger
          )
        ),
        getPromiseValue('Alerts', getAlertsStats(collectorFetchContext.esClient, logger)),
      ]);

      return {
        indices,
        accounts_stats,
        resources_stats,
        rules_stats,
        installation_stats,
        alerts_stats,
      };
```

Reply:
and we can replace `allSettled` with `all`, because we wrap it with a handler that keeps it safe